Add formal verification script for Surelog vs Yosys parsers with logs and blacklist

mandrys opened the pull request:
Originally submitted by @alaindargelas in https://github.com/antmicro/yosys-systemverilog/pull/1601

mglb commented:
What are these .tcl files for? They seem unused.
We probably should modify our formal verification test to do what those .tcl do.
`gold.v` files shouldn't be verified in a way we do it right now. They should be used in place of `yosys_gate.v` (which shouldn't be generated).

mglb reviewed (changes requested):
What are these .tcl files for? They seem unused.
We probably should modify our formal verification test to do what those .tcl do.
`gold.v` files shouldn't be verified in a way we do it right now. They should be used in place of `yosys_gate.v` (which shouldn't be generated).

mandrys commented:
I removed the tcl files and modified our formal verification flow to test against `gold.v` files when they are present and not to generate `yosys_gate.v` then.

mandrys commented:
I modified the formal verification flow again as what we expect from it in case of _bsg_micro_design_results_ test set, is to run formal verification on `gold.v` files.
These `gold.v` files are the reference for comparison with `dut.v` files, and `dut.v` files are generated by the plugin (read by systemverilog plugin, and written with write_verilog).
Additionally, we'd like to prepare another job that generates `dut.v` files again and compares them to the ones commited to the repo, but this could be done in a separate PR.

mglb commented on run_fv_tests.mk:
```diff
@@ -78,7 +80,9 @@ help :
 ifeq (${help_only},0)
 
-rel_v_files := $(shell cd ${TEST_SUITE_DIR}; echo */**/*.{v,sv})
+dut_to_gold_rename_key := bsg_micro_designs_results
+
+rel_v_files := $(call rename_dut_to_gold,$(shell cd ${TEST_SUITE_DIR}; echo */**/*.{v,sv}),${dut_to_gold_rename_key})
```
Since we want to test `gold.v` as well (i.e. not only use it as a reference), let's just add all `dut.v` to skiplist.txt for now (with a comment why) and remove extra code from the run_fv_tests.mk.

mandrys replied:
Done.

mglb commented:
bsg tests are not tested in formal verification currently, so this PR is no longer applicable.
